test_scripts/Resumption/Handling_errors_from_HMI/028_resume_all_data_with_one_error_unsubscribeButton_unexpected_disconnect.lua

```diff
@@ -0,0 +1,124 @@
+--
+-- Requirement summary:TBD
+--
+-- Description:
+-- In case:
+-- 1. ButtonSubscription and other resumption data are added by app
+-- 2. Unexpected disconnect and reconnect are performed
+-- 3. App reregisters with actual HashId
+-- 4. OnButtonSubscription(isSubscribed=true) notification is sent from SDL to HMI during resumption
+-- 5. Rpc_n is requested from SDL
+-- 6. HMI responds with error resultCode to Rpc_n request
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. remove already restored data and send OnButtonSubscription(isSubscribed=false) to HMI
+-- 3. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Common Functions ]]
+local function checkResumptionDataWithErrorResponse(pAppId, pErrorResponceRpc, pErrorResponseInterface)
+  local rpcsRevertLocal = common.cloneTable(common.rpcsRevert)
+  if pErrorResponceRpc == "addCommand" and pErrorResponseInterface == "VR" then
+    rpcsRevertLocal.addCommand.VR = nil
+    common.getHMIConnection():ExpectRequest("VR.AddCommand")
+    :Do(function(_, data)
+        if data.params.type == "Command" then
+          common.sendResponseWithDelay(data)
+        else
+          common.sendResponse(data)
+        end
+        if data.params.type == "Choice" then
+          common.removeData.DeleteVRCommand(pAppId, data.params.type, 1)
+        end
+      end)
+    :Times(2)
+  elseif pErrorResponceRpc == "createIntrerationChoiceSet" then
+    rpcsRevertLocal.addCommand.VR = nil
+    common.getHMIConnection():ExpectRequest("VR.AddCommand")
+    :Do(function(_, data)
+        if data.params.type == "Choice" then
+          common.sendResponseWithDelay(data)
+        else
+          common.sendResponse(data)
+        end
+        if data.params.type == "Command" then
+          common.removeData.DeleteVRCommand(pAppId, data.params.type, 1)
+        end
+      end)
+    :Times(2)
+  else
+    rpcsRevertLocal[pErrorResponceRpc][pErrorResponseInterface] = nil
+    common.getHMIConnection():ExpectRequest(common.getRpcName(pErrorResponceRpc, pErrorResponseInterface))
+    :Do(function(_, data)
+        common.sendResponseWithDelay(data, pErrorResponseInterface, pErrorResponseInterface)
+      end)
+  end
+
+  for k, value in pairs (rpcsRevertLocal) do
+    if common.resumptionData[pAppId][k] then
+      for interface in pairs (value) do
+        rpcsRevertLocal[k][interface](pAppId)
+      end
+    end
+  end
+
+  local customButton = 0
+  local buttonSubscribed = 0
+  local buttonUnsubscribed = 0
```

Comment:
@GetmanetsIrina Since `customButton`, `buttonSubscribed` and `buttonUnsubscribed` are flags it's better to make them `boolean` and rename to something like `is...`